var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -37,61 +42,67 @@ class ModelHippocampus(SimModel):
 
     phases = ('build', 'install')
 
-    mech_name = ''
+    mech_name = None
     """The name of the mechanism, defined in subclasses"""
 
     def build(self, spec, prefix):
-        """Build phase
-        """
-        link_flag = '-Wl,-rpath,' + prefix.lib
-        self._build_mods('mod', link_flag)
+        """Build phase"""
+        self._build_mods('mod')
+
+    @property
+    def lib_suffix(self):
+        return ('_' + self.mech_name) if self.mech_name else ''
 
     def _build_mods(self, mods_location, link_flag='', include_flag='', corenrn_mods=None):
         """Build shared lib & special from mods in a given path
         """
-        profile_flag = ' -DENABLE_TAU_PROFILER' if '+profile' in self.spec else ''
-        include_flag += profile_flag
+        # pass include and link flags for all dependency libraries
+        # Compiler wrappers are not used to have a more reproducible building
+        dep_names = set(self.spec.dependencies_dict('link').keys())
+        for dep in dep_names:
+            if self.spec[dep].prefix in ('/usr', '/usr/local'):
+                raise Exception('Dependency lib "' + dep + '" coming from system prefix')
+        link_flag += ' ' + ' '.join(self.spec[dep].libs.ld_flags for dep in dep_names)
+        link_flag += ' ' + ' '.join(self.spec[dep].libs.rpath_flags for dep in dep_names)
```

Comment:
Is this not indented correctly? (should be inside loop?)

Reply:
we could use a plain loop, it's a matter of preference (notice the `for` inside the join)

Reply:
Changed to a plain loop